format/spec.md

```diff
@@ -1237,17 +1237,37 @@ Content file (data or delete) is serialized as a JSON object according to the fo
 | **`equality-ids`**       |`JSON list of int: Field ids used to determine row equality in equality delete files`|`[1]`|
 | **`sort-order-id`**      |`JSON int`|`1`|
 
-### File Scan Task Serialization
-
-File scan task is serialized as a JSON object according to the following table.
-
-| Metadata field       |JSON representation|Example|
-|--------------------------|--- |--- |
-| **`schema`**          |`JSON object`|`See above, read schemas instead`|
-| **`spec`**            |`JSON object`|`See above, read partition specs instead`|
-| **`data-file`**       |`JSON object`|`See above, read content file instead`|
-| **`delete-files`**    |`JSON list of objects`|`See above, read content file instead`|
-| **`residual-filter`** |`JSON object: residual filter expression`|`{"type":"eq","term":"id","value":1}`|
+### Task Serialization
+
+There could be different implementations of tasks, 
+e.g., `BaseFileScanTask` and `StaticDataTask` in Java.
+A enum `task-type` field is needed to distinguish different task types.
```

Comment:
internally `task-type` can be an enum, but I don't think we need to mention this here, since the JSON field is a string. Maybe just say **A `task-type` field is used to distinguish between different task types.**